packages/typescript-estree/src/convert.ts

```diff
@@ -690,6 +690,10 @@ export class Converter {
     typeArguments: ts.NodeArray<ts.TypeNode>,
     node: TSESTreeToTSNode<TSESTree.TSTypeParameterInstantiation>,
   ): TSESTree.TSTypeParameterInstantiation {
+    if (typeArguments.length === 0) {
```

Comment:
I'm not sure but does this deal with `function f<>() {}`? I don't see a test. And of course `type T<> = any` and `class C<> {}`

Reply:
Looks like you want to update `convertTSTypeParametersToTypeParametersDeclaration` too